CMakeLists.txt

```diff
@@ -43,6 +43,19 @@ option(ASM "Use assembly routines." ON)
 cmake_dependent_option(CXX20 "Enable compilation in C++20 mode." OFF "NOT MSVC" ON)
 option(THREADLOCAL "Enable features that depend on the C++ thread_local keyword (currently just thread names in debug logs)." ON)
 
+# TODO: These tri-state options will be removed and most features
+#       will become opt-in by default before merging into master.
```

Comment:
What does "opt-in by default" mean? That there will be just two options: ON and OFF, that it will be ON by default and (it follows that) if the corresponding library is not found, then it will result in an error?

Reply:
> What does "opt-in by default" mean? .... that it will be ON by default

No. It means that basically all features/options will be OFF by default, and builders will have to actively opt-in to what they want. If they opt-in to something, and do not have the required libs/prerequisites, the build will error.